File: backend/profile-service/src/main/java/com/example/profileservice/api/pub/ProfilePublicController.java
package com.example.profileservice.api.pub;

import com.example.profileservice.profile.dto.CreateProfileDto;
import com.example.profileservice.profile.dto.ProfileDto;
import com.example.profileservice.profile.service.ProfileService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/** 안드로이드가 게이트웨이를 거쳐 부르는 프로필 API. */
@RestController
@RequiredArgsConstructor
@RequestMapping("/api-public/profile")
public class ProfilePublicController {

    private final ProfileService profileService;

    @GetMapping("/{memberId}/{id}")
    public ResponseEntity<ProfileDto> getProfile(@PathVariable("memberId") String memberId, @PathVariable("id") String id) {
        return ResponseEntity.ok().body(profileService.getMemberProfile(memberId, id));
    }

    @GetMapping("/{memberId}")
    public ResponseEntity<List<ProfileDto>> getProfiles(@PathVariable("memberId") Long memberId) {
        return ResponseEntity.ok().body(profileService.getMemberProfiles(memberId));
    }

    @GetMapping("/latest/{memberId}")
    public ResponseEntity<ProfileDto> getLatestProfile(@PathVariable("memberId") String memberId) {
        return ResponseEntity.ok().body(profileService.getMemberLatestProfile(memberId));
    }

    @GetMapping("/{memberId}/{id}/{count}")
    public ResponseEntity<List<ProfileDto>> getProfilesOffset(@PathVariable("memberId") String memberId, @PathVariable("id") String id, @PathVariable("count") String count) {
        return ResponseEntity.ok().body(profileService.getMemberProfileOffset(memberId, id, count));
    }

    @GetMapping("/total/count/{memberId}")
    public ResponseEntity<Long> getTotalProfileCount(@PathVariable("memberId") String memberId) {
        return ResponseEntity.ok().body(profileService.getMemberProfileTotalCount(memberId));
    }

    @PostMapping
    public ResponseEntity<ProfileDto> createProfile(@RequestBody CreateProfileDto createProfileDto) {
        return ResponseEntity.ok().body(profileService.registerProfile(createProfileDto));
    }

    @DeleteMapping("/{memberId}/{id}")
    public ResponseEntity<Long> removeProfile(@PathVariable("memberId") String memberId, @PathVariable("id") String id) {
        return ResponseEntity.ok().body(profileService.deleteProfile(memberId, id));
    }
}

File: backend/profile-service/src/main/kotlin/com/example/profileservice/api/pub/ProfilePublicController.kt
package com.example.profileservice.api.pub

import com.example.profileservice.profile.dto.CreateProfileDto
import com.example.profileservice.profile.dto.ProfileDto
import com.example.profileservice.profile.service.ProfileService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/** 안드로이드가 게이트웨이를 거쳐 부르는 프로필 API. */
@RestController
@RequestMapping("/api-public/profile")
class ProfilePublicController(private val profileService: ProfileService) {

    @GetMapping("/{memberId}/{id}")
    fun getProfile(@PathVariable("memberId") memberId: String, @PathVariable("id") id: String): ResponseEntity<ProfileDto> =
        ResponseEntity.ok().body(profileService.getMemberProfile(memberId, id))

    @GetMapping("/{memberId}")
    fun getProfiles(@PathVariable("memberId") memberId: Long): ResponseEntity<List<ProfileDto>> =
        ResponseEntity.ok().body(profileService.getMemberProfiles(memberId))

    @GetMapping("/latest/{memberId}")
    fun getLatestProfile(@PathVariable("memberId") memberId: String): ResponseEntity<ProfileDto> =
        ResponseEntity.ok().body(profileService.getMemberLatestProfile(memberId))

    @GetMapping("/{memberId}/{id}/{count}")
    fun getProfilesOffset(
        @PathVariable("memberId") memberId: String,
        @PathVariable("id") id: String,
        @PathVariable("count") count: String,
    ): ResponseEntity<List<ProfileDto>> = ResponseEntity.ok().body(profileService.getMemberProfileOffset(memberId, id, count))

    @GetMapping("/total/count/{memberId}")
    fun getTotalProfileCount(@PathVariable("memberId") memberId: String): ResponseEntity<Long> =
        ResponseEntity.ok().body(profileService.getMemberProfileTotalCount(memberId))

    @PostMapping
    fun createProfile(@RequestBody createProfileDto: CreateProfileDto): ResponseEntity<ProfileDto> =
        ResponseEntity.ok().body(profileService.registerProfile(createProfileDto))

    @DeleteMapping("/{memberId}/{id}")
    fun removeProfile(@PathVariable("memberId") memberId: String, @PathVariable("id") id: String): ResponseEntity<Long> =
        ResponseEntity.ok().body(profileService.deleteProfile(memberId, id))
}
